javascriptv3/example_code/iam/scenarios/src/iam_basics.js

```diff
@@ -0,0 +1,410 @@
+            Path: "/",
+            RoleName: role_name // The name of the new role.
+          };
+          const data = await iamClient.send(new CreateRoleCommand(role_params));
+          console.log("Success. Role created. Role Arn: ", data.Role.Arn);
+          const role_arn = data.Role.Arn;
+          try {
+            console.log(
+                "\nAttaching to the role the policy with permissions to list all buckets....\n"
+            );
+            const params = {
+              PolicyArn: s3_policy_arn,
+              RoleName: role_name,
+            };
+            await iamClient.send(new AttachRolePolicyCommand(params));
+            console.log("Success. Policy attached successfully to role.");
+            try {
+              console.log(
+                  "\nCreate a policy that enables the user to assume the role ....\n"
+              );
+              const myNewPolicy = {
+                Version: "2012-10-17",
+                Statement: [
+                  {
+                    Effect: "Allow",
+                    Action: ["sts:AssumeRole"],
+                    Resource: role_arn,
+                  },
+                ],
+              };
+              const policy_params = {
+                PolicyDocument: JSON.stringify(myNewPolicy),
+                PolicyName: assume_policy_name,
+              };
+              const data = await iamClient.send(
+                  new CreatePolicyCommand(policy_params)
+              );
+              console.log(
+                  "Success. Policy created. Policy ARN: " + data.Policy.Arn
+              );
+
+              const assume_policy_arn = data.Policy.Arn;
+              try {
+                console.log("\nAttaching the policy to the user.....\n");
+                const attach_policy_to_user_params = {
+                  PolicyArn: assume_policy_arn,
+                  UserName: user_name,
+                };
+                const data = await iamClient.send(
+                    new AttachUserPolicyCommand(attach_policy_to_user_params)
+                );
+                console.log(
+                    "\nWaiting 10 seconds for policy to be attached...\n"
+                );
+                wait(10000);
+                console.log(
+                    "Success. Policy attached to user " + user_name + "."
+                );
+                try {
+                  console.log(
+                      "\nAssume for the user the role with permission to list all buckets....\n"
+                  );
+                  const assume_role_params = {
+                    RoleArn: role_arn, //ARN_OF_ROLE_TO_ASSUME
+                    RoleSessionName: "session1",
+                    DurationSeconds: 900,
+                  };
+                  // Create Amazon STS client with the credentials for the user. Remember, the user has permissions to assume roles using Amazon STS.
```

Comment:
// Create an AWS STS client with the credentials for the user. Remember, the user has permissions to assume roles using AWS STS.

Reply:
done